lib/functions/general/apt-utils.sh

```diff
@@ -66,8 +66,25 @@ function apt_find_upstream_package_version_and_download_url() {
 	if [[ "${found_package_filename}" == "${sought_package_name}_"* ]]; then
 		display_alert "Found upstream base-files package filename" "${found_package_filename}" "info"
 	else
-		display_alert "Could not find package filename for '${sought_package_name}' in '${package_info_download_urls[*]}'" "looking for ${sought_package_name}" "warn"
-		return 1
+		display_alert "Could not find package filename for '${sought_package_name}' in '${package_info_download_urls[*]}'" "Trying alternative method to get ${sought_package_name}" "warn"
+		# Try alternative method since packages.debian.org is down often
+		# Use -N with wget so it always downloads the latest file, overwriting the local one if it exists
+		run_host_command_logged wget --no-verbose -N https://${mirror_with_slash}/dists/${RELEASE}/main/binary-${ARCH}/Packages.xz
```

Comment:
Yeah I'm pretty certain this will break under multiple simultaneous executions, as it writes to the "current directory", and that might be the same directory when run in the prepare json matrix pipeline job.  Ideally this should be done without touching the disk, by using a series of pipes. Those parallel jobs are run with `CONFIG_DEFS_ONLY` and other wide-affecting flags too, so beware.

Reply:
The "real" solution to this is to move it out of band; implement the gathering of information (incl base-files version and references, but also other info about the RELEASE+ARCH) to a scheduled, frequent, GitHub Actions job. We'd get all the info required, massage and process it into a format for easy armbian/build consumption, and publish it in a git repo. Then armbian/build would only fetch from that repo -- a mechanism we already do for many of the packages.

Reply:
@rpardini Is the "real" solution something being planned / worked on at the moment ? 

PS: As a image build user, I can't build anything when ubuntu / debian official sites act up and it has been a lot lately :-( and it is not armbian fault of course :-)